agent/discovery/collector/client.go

```diff
@@ -64,10 +65,15 @@ func NewCollectorClient(config *Config) (*client, error) {
 
 	machineID := strings.TrimSpace(string(machineIDBytes))
 
+	agentID, err := uuid.NewRandomFromReader(strings.NewReader(machineID))
+	if err != nil {
```

Comment:
Is the UUID Random?
In this case what happens if the agent or the machine is restarted?

We need to find a way to have deterministic IDs, e.g. namespacing could help

Reply:
From what I could see, this line returns always the same UUID given the same machineID.

I can double check.

Reply:
I think this is a seed for a random number so yes, given the same seed the random number should be the same.

Reply:
I'll add a test for this.

```
Pseudocode

loop n times and call
uuid.NewRandomFromReader(strings.NewReader(machineID))

check all the uuids are the same
```

Would this help?

Reply:
I would still use a namespace for this one rather than the random  uuid

Reply:
I;ll give it a try :wink:

Reply:
@fabriziosestito changed, please take a look.